### PR TITLE
Update workflow parameters to use camel case

### PR DESCRIPTION
**Pull request by tsurdilo**

### CODE REVIEW

**Comment by ruromero** [comment]
Thanks for this change. I think it is more aligned with JSON naming conventions and best practices.

**Comment by tsurdilo** [comment]
@ruromero thanks for review. updated (without force pushing even ;) )

File: workflow/spec/spec.md [hunk 1]
```diff
@@ -908,7 +908,7 @@ In addition, states outside a Parallel state cannot transition to a state within
 The Parallel state generates an output array in which each element is the output for a branch. 
 The elements of the output array need not be of the same type.
 
-The "wait-for-completion" property allows the parallel state to manage branch executions. If this flag is set to 
+The "waitForCompletion" property allows the parallel state to manage branch executions. If this flag is set to 
 true, the branches parallel parent state must wait for this branch to finish before continuing execution.
```
Comment (cathyhongzhang) [inline workflow/spec/spec.md]: LGTM

**Comment by ruromero** [comment]
@tsurdilo sorry but can you rebase again before merging?

**Comment by tsurdilo** [comment]
will rebase and merge (2 LGTM)

**Pull request merged**